python-checks/src/main/java/org/sonar/python/checks/TrailingCommentCheck.java

```diff
@@ -41,23 +46,56 @@ public class TrailingCommentCheck extends PythonSubscriptionCheck {
 
   private int previousTokenLine;
 
+  private List<String> lines;
+
   @Override
   public void initialize(Context context) {
     Pattern pattern = Pattern.compile(legalCommentPattern);
-    context.registerSyntaxNodeConsumer(Tree.Kind.FILE_INPUT, ctx -> previousTokenLine = -1);
+    context.registerSyntaxNodeConsumer(Tree.Kind.FILE_INPUT, ctx -> {
+      previousTokenLine = -1;
+      lines = ctx.pythonFile().content().lines().collect(Collectors.toList());
+    });
 
     context.registerSyntaxNodeConsumer(Tree.Kind.TOKEN, ctx -> {
-      Token pyToken = (Token) ctx.syntaxNode();
-      for (Trivia trivia : pyToken.trivia()) {
+      Token token = (Token) ctx.syntaxNode();
+      for (Trivia trivia : token.trivia()) {
         if (previousTokenLine == trivia.token().line()) {
           String comment = trivia.token().value();
           if (!pattern.matcher(comment).matches()) {
-            ctx.addIssue(trivia.token(), MESSAGE);
+            IssueWithQuickFix issue = (IssueWithQuickFix) ctx.addIssue(trivia.token(), MESSAGE);
+            addQuickFix(issue, trivia.token());
           }
         }
       }
-      previousTokenLine = pyToken.line();
+      previousTokenLine = token.line();
     });
   }
+
+  private void addQuickFix(IssueWithQuickFix issue, Token commentToken) {
+    String line = lines.get(commentToken.line() - 1);
+    String indent = calculateIndent(line);
+    PythonTextEdit insert = new PythonTextEdit(
+      indent + commentToken.value() + "\n",
+      commentToken.line(), 0,
+      commentToken.line(), 0);
```

Comment:
```suggestion
    PythonTextEdit insert = PythonTextEdit.insertAtPosition(commentToken.line(), 0, indent + commentToken.value() + "\n");
```
`PythonTextEdit.insertAtPosition` is private right now, but could reduce the repetition here if made public.